ios/RNSScreenComponentView.mm

```diff
@@ -115,6 +119,54 @@ - (void)notifyDisappear
   }
 }
 
+- (void)setGestureEnabled:(BOOL)gestureEnabled
+{
+#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && defined(__IPHONE_13_0) && \
+    __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_13_0
+  if (@available(iOS 13.0, tvOS 13.0, *)) {
+    _controller.modalInPresentation = !gestureEnabled;
+  }
+#endif
+  _gestureEnabled = gestureEnabled;
+}
+
+- (BOOL)isMountedUnderScreenOrReactRoot
+{
+  for (UIView *parent = self.superview; parent != nil; parent = parent.superview) {
+    if ([parent isKindOfClass:[RCTRootComponentView class]] || [parent isKindOfClass:[RNSScreenComponentView class]]) {
+      return YES;
+    }
+  }
+  return NO;
+}
+
+- (void)didMoveToWindow
+{
+  if (self.window != nil && ![self isMountedUnderScreenOrReactRoot]) {
```

Comment:
It was added for proper gesture handling in modals and it would be good to check if it works correctly there after we merge modal `stackPresentation`s. It will be good to test on example provided in https://github.com/software-mansion/react-native-screens/pull/784